src/Config/ValueProvider/TestFrameworkConfigPathProvider.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Copyright © 2017 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+declare(strict_types=1);
+
+namespace Infection\Config\ValueProvider;
+
+use Infection\Config\ConsoleHelper;
+use Infection\Config\Guesser\PhpUnitPathGuesser;
+use Infection\TestFramework\Config\TestFrameworkConfigLocator;
+use Infection\TestFramework\TestFrameworkTypes;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+
+class TestFrameworkConfigPathProvider
+{
+    /**
+     * @var TestFrameworkConfigLocator
+     */
+    private $testFrameworkConfigLocator;
+    /**
+     * @var ConsoleHelper
+     */
+    private $consoleHelper;
+
+    /**
+     * @var QuestionHelper
+     */
+    private $questionHelper;
+
+    public function __construct(TestFrameworkConfigLocator $testFrameworkConfigLocator, ConsoleHelper $consoleHelper, QuestionHelper $questionHelper)
+    {
+        $this->testFrameworkConfigLocator = $testFrameworkConfigLocator;
+        $this->consoleHelper = $consoleHelper;
+        $this->questionHelper = $questionHelper;
+    }
+
+    public function get(InputInterface $input, OutputInterface $output, array $dirsInCurrentDir, string $testFramework): ?string
+    {
+        try {
+            return $this->testFrameworkConfigLocator->locate($testFramework);
+        } catch (\Exception $e) {
+            if ($testFramework !== TestFrameworkTypes::PHPUNIT) {
```

Comment:
just a question: does it make sense to combine this condition with the next one?

```php
if ($testFramework !== TestFrameworkTypes::PHPUNIT || !file_exists('composer.json')) {
   // ...
```

Reply:
If you combine them, you need to write additional tests to kill mutants.
And I think it's more readable to just split them and fast return.